### PR TITLE
having a single Buffer in LoggerSet for locking

### PR DESCRIPTION
**Pull request by kazu-yamamoto**
This should fix #196

### CODE REVIEW

**Comment by kazu-yamamoto** [comment]
@ruicc @khibino Would you review this PR?

**Review by khibino — APPROVED** [review]
LGTM

**Comment by kazu-yamamoto** [comment]
@ruicc Are you OK with this PR?

**Comment by ruicc** [comment]
I will check behaviours..

File: fast-logger/System/Log/FastLogger/Logger.hs [hunk 1]
```diff
@@ -21,20 +20,19 @@ import System.Log.FastLogger.LogStr
...
-                             <*> newIORef mempty
+newLogger :: IO Logger
+newLogger = Logger <$> newIORef mempty
 
 ----------------------------------------------------------------
 
-pushLog :: IORef FD -> Logger -> LogStr -> IO ()
-pushLog fdref logger@(Logger size mbuf ref) nlogmsg@(LogStr nlen nbuilder)
+pushLog :: IORef FD -> BufSize -> MVar Buffer -> Logger -> LogStr -> IO ()
+pushLog fdref size mbuf logger@(Logger ref) nlogmsg@(LogStr nlen nbuilder)
```
Comment (ruicc) [inline fast-logger/System/Log/FastLogger/Logger.hs:33]: Writing is protected by the same `MVar` when message size is longer than buffer size. OK.

**Review by ruicc — APPROVED** [review]
Checking code and behaviours finished. LGTM!

**Comment by kazu-yamamoto** [comment]
Merged.
This should be version 3.1.0.

**Pull request merged**